assets/blueprints/genesis_helper/src/lib.rs

```diff
@@ -4,178 +4,204 @@ use scrypto::blueprints::epoch_manager::*;
+                rounds_per_epoch,
+                xrd_vault: Vault::with_bucket(whole_lotta_xrd),
+                resource_vaults: KeyValueStore::new(),
+                validators: KeyValueStore::new(),
             }
+            .instantiate()
+            .globalize()
+        }
 
-            // Create the epoch manager with initial validator set...
-            let mut validators_with_initial_stake = vec![];
-            for (validator_idx, validator) in genesis_data.validators.into_iter().enumerate() {
-                let initial_stake_amount = genesis_data
-                    .stakes
-                    .get(&validator_idx)
-                    .map(|stakes| {
-                        stakes
-                            .iter()
-                            .map(|(_, xrd_amount)| xrd_amount)
-                            .cloned()
-                            .sum()
-                    })
-                    .unwrap_or(Decimal::ZERO);
-                let initial_stake_bucket = whole_lotta_xrd.take(initial_stake_amount);
-                validators_with_initial_stake.push((
-                    validator.key,
-                    validator.component_address,
-                    initial_stake_bucket,
-                ));
+        pub fn ingest_data_chunk(&mut self, chunk: GenesisDataChunk) {
+            match chunk {
+                GenesisDataChunk::Validators(validators) => self.create_validators(validators),
+                GenesisDataChunk::Stakes {
+                    accounts,
+                    allocations,
+                } => self.allocate_stakes(accounts, allocations),
+                GenesisDataChunk::Resources(resources) => self.create_resources(resources),
+                GenesisDataChunk::ResourceBalances {
+                    accounts,
+                    allocations,
+                } => self.allocate_resources(accounts, allocations),
+                GenesisDataChunk::XrdBalances(allocations) => self.allocate_xrd(allocations),
+            }
+        }
+
+        fn create_validators(&mut self, validators: Vec<GenesisValidator>) {
+            for validator in validators.into_iter() {
+                self.create_validator(validator);
             }
+        }
 
-            let lp_buckets: Vec<Bucket> = Runtime::call_function(
-                EPOCH_MANAGER_PACKAGE,
-                EPOCH_MANAGER_BLUEPRINT,
-                EPOCH_MANAGER_CREATE_IDENT,
-                scrypto_encode(&EpochManagerCreateInput {
-                    validator_owner_token,
-                    component_address: epoch_manager_component_address,
-                    validator_set: validators_with_initial_stake,
-                    initial_epoch,
-                    rounds_per_epoch,
-                    num_unstake_epochs,
+        fn create_validator(&mut self, validator: GenesisValidator) {
+            let (validator_address, owner_token_bucket): (ComponentAddress, Bucket) =
+                Runtime::call_method(
```

Comment:
Can we write/use sdks for epoch_manager and accounts rather than using `Runtime::call_method`

Reply:
done